src/transaction_builder.js

```diff
@@ -14,9 +14,20 @@ var Transaction = require('./transaction')
 
 // inspects a scriptSig w/ optional redeemScript and
 // derives any input information required
-function expandInput (scriptSig, redeemScript) {
-  var scriptSigChunks = bscript.decompile(scriptSig)
-  var prevOutType = bscript.classifyInput(scriptSigChunks, true)
+function expandInput (scriptSig, redeemScript, witnessStack) {
+  var witnessType
+  if (witnessStack) {
+    witnessType = bscript.classifyWitness(witnessStack)
+  }
+
+  var prevOutType, scriptSigChunks
+  if (scriptSig.length === 0 && witnessStack) {
```

Comment:
ScriptSig's aren't empty if the transaction is P2SH | P2WPKH - the scriptSig would be a push of the `[OP_0 [20 byte hash]]`